docs/environment.yml

```diff
@@ -2,7 +2,8 @@ name: readthedocs
 channels:
   - conda-forge
 dependencies:
-  - python=3.12
-  - sphinx_rtd_theme=2.0.*
-  - sphinxcontrib-bibtex=2.6.*
+  - linkchecker 10.4.*
+  - python >=3.9,<3.13
+  - sphinx_rtd_theme 3.0.*
+  - sphinxcontrib-bibtex 2.6.*
```

Comment:
A downside to pinning to a single Python version, as was added recently, is that a developer's dev shell may have a different, but completely acceptable, Python version (one compatible with the run requirement specified in `meta.yaml`), which will then be up or downgraded to the version specified here, along with potentially many other packages. By providing a range of acceptable Python versions that matches the run requirement, it increases the chances that the specs here will lead to only installation of missing packages, without up/downgrades.

I think there's still a change that an up/downgrade could happen: If e.g. a user had a Python 3.9 dev 
 shell and one of the non-`python` packages pinned here wasn't compatible with 3.9, then a Python upgrade could happen. But hopefully most packages will be compatible with all currently-supported Python releases. `uwtools` aims to be, but I don't know how universal that might be in the package ecosystem.